### PR TITLE
feat(form-pill-group): update designs and add error variant

### PR DESCRIPTION
**Pull request by TheSisb**
### Changes in this PR
**FormPillGroup**
  * Update components to match the new designs
  * Add a error variant
  * Support disabled pills
  * Re-organize stories to make all the variants and states clearer
  * Update the tests to work with the updated components

**Box**
  * add _focus_hover pseudo-selector
  * add missing disabled prop

**Design tokens**
  * add color-background-error-strongest and color-text-error-stronger tokens
  * update snapshot tests to include the new tokens

**Website**
  * Update the display pill group page to match the new spec
  * Update the form pill group page to match the new spec

### Open questions
1. How do we feel about [the form pill group props table](https://deploy-preview-2526--paste-docs.netlify.app/components/form-pill-group#props)? Should I update the markdown so all tables horizontally scroll?

### CODE REVIEW

File: packages/paste-core/components/form-pill-group/src/FormPill.styles.ts
Comment (TheSisb) [inline packages/paste-core/components/form-pill-group/src/FormPill.styles.ts:87]: TODO: 
- Review and fix the text color on hover for the error state. It should use the new token listed in the ticket on the Close icon and the FormPillButton content too.
- When focusing and hovering, the border shouldn't be visible. I had to create a new pseudoProp on Box for this called `_focus_hover`, but I'm not sure it's working as I didn't have time to test it yet. I added it into line 41 in this file just in case it works on the first try.

File: packages/paste-core/components/form-pill-group/src/FormPillButton.tsx
Comment (TheSisb) [inline packages/paste-core/components/form-pill-group/src/FormPillButton.tsx:64]: I set a solid height here because this height needs to match the `PillCloseIcon.tsx`component height and width. If we can use a new sizing token here, like `sizePillHeight` or something, that might be best as it would render in `rem` units instead of `px` units. This would allow the component to resize correctly when resizing font size at the browser level.

File: packages/paste-core/components/form-pill-group/src/PillCloseIcon.tsx
Comment (TheSisb) [inline packages/paste-core/components/form-pill-group/src/PillCloseIcon.tsx:45]: Note: I normally don't like double ternaries. In fact I think there's an ESLint rule to prevent these. However I was iterating rapidly and didn't want to keep shuffling if/else logic around. This could use a second pair of eyes once designs are solidified.

File: packages/paste-core/components/form-pill-group/stories/customization.stories.tsx
Comment (TheSisb) [inline packages/paste-core/components/form-pill-group/stories/customization.stories.tsx:1]: I like moving Customization stories into their own sub-story folder and file because it keeps things clean in the storybook sidebar. Hence this change.

File: packages/paste-core/components/form-pill-group/stories/index.stories.tsx
Comment (TheSisb) [inline packages/paste-core/components/form-pill-group/stories/index.stories.tsx:44]: TODO: I need to rethink these story names and which ones display errors, disabled, avatars, etc. to show all the new permutations cleanly.

File: packages/paste-core/primitives/box/package.json
Comment (TheSisb) [inline packages/paste-core/primitives/box/package.json:53]: Note: we should add this everywhere so we can build directly from within a single package. That's why I added it here. Won't require a changeset because it's just for us.

File: packages/paste-core/primitives/box/src/PseudoPropStyles.ts
Comment (TheSisb) [inline packages/paste-core/primitives/box/src/PseudoPropStyles.ts:9]: This change requires a changeset.

Should work, but untested in Paste. Only in codesandbox: https://codesandbox.io/s/bold-dust-10j4or?file=/src/styles.css

File: packages/paste-core/primitives/box/src/types.ts
Comment (TheSisb) [inline packages/paste-core/primitives/box/src/types.ts:93]: This should be mentioned in a changeset too. Box should be able to accept the disabled prop if used as an input type or button element

File: packages/paste-core/components/form-pill-group/src/PillCloseIcon.tsx
Comment (SiTaggart) [inline packages/paste-core/components/form-pill-group/src/PillCloseIcon.tsx:57]: question: So for here, given this is an icon, though not a fully blown paste icon, why wouldn't we be using a icon size e.g. `$size-icon-40` (24px).

Then using that as our reference for a height we need to align to?

Comment (shleewhite) [reply packages/paste-core/components/form-pill-group/src/PillCloseIcon.tsx:57]: Oh, that works! Will use the existing token.

**Review by TheSisb — COMMENTED** [review]
✅

File: packages/paste-core/components/form-pill-group/src/FormPillButton.tsx
Comment (gloriliale) [inline packages/paste-core/components/form-pill-group/src/FormPillButton.tsx:11]: Suggestion(non-blocking): for consistency, it might be better to use `element?: BoxElementProps['element']` here.

File: packages/paste-website/src/component-examples/FormPillGroup.tsx
Comment (gloriliale) [inline packages/paste-website/src/component-examples/FormPillGroup.tsx:254]: Todo: at least one (or all?) of these `FormPill`s need to include the `disabled` prop in this example

File: packages/paste-website/src/pages/components/display-pill-group/index.mdx
Comment (gloriliale) [inline packages/paste-website/src/pages/components/display-pill-group/index.mdx:4]: issue: this change in description isn't showing up in the netlify preview. Not sure why, since the rest of the changes seem to show up just fine 🤔

File: packages/paste-website/src/pages/components/form-pill-group/index.mdx
Comment (gloriliale) [inline packages/paste-website/src/pages/components/form-pill-group/index.mdx:4]: issue: same thing as in display pill group- preview of the site doesn't reflect this change.

File: packages/paste-website/src/pages/components/form-pill-group/index.mdx
Comment (gloriliale) [inline packages/paste-website/src/pages/components/form-pill-group/index.mdx:159]:
```suggestion
Interaction states on Form Pills need to be managed separately as shown in the following examples by using the `useFormPillState` hook. The returned state should be spread onto each component. This will provide you with some internal state management and keyboard navigation.
```

Suggestion(non-blocking): using "the examples after this one" made me think that this example wouldn't use it, but it does

**Review by gloriliale — COMMENTED** [review]
Yahoo for this being almost done! Left some comments; so excited by how it looks!

File: .changeset/hungry-plants-repair.md
Comment (gkassym) [reply .changeset/hungry-plants-repair.md:6]: Hi @nkrantz, we will not publish the extension after the PR merging, so my next step is to use tokens from paste dist.

File: packages/paste-website/src/pages/components/display-pill-group/index.mdx
Comment (shleewhite) [reply packages/paste-website/src/pages/components/display-pill-group/index.mdx:4]: Ooops, forgot to put it in the body of the page. Should be fixed now.

File: packages/paste-website/src/component-examples/FormPillGroup.tsx
Comment (shleewhite) [reply packages/paste-website/src/component-examples/FormPillGroup.tsx:254]: Ack! Thanks for catching that. Fixed.

File: .changeset/real-moles-exercise.md
Comment (shleewhite) [reply .changeset/real-moles-exercise.md:3]: Hmmm... I think patch is ok... These are 2 really small props that people probably assumed were already available.

File: packages/paste-core/components/form-pill-group/src/FormPill.styles.ts
Comment (SiTaggart) [inline packages/paste-core/components/form-pill-group/src/FormPill.styles.ts:160]: praise: nice, I like these collected together like this

File: packages/paste-core/components/form-pill-group/src/FormPill.tsx
Comment (SiTaggart) [inline packages/paste-core/components/form-pill-group/src/FormPill.tsx:61]: praise: nice, good guard rail

File: packages/paste-core/components/form-pill-group/stories/customization.stories.tsx
Comment (SiTaggart) [inline packages/paste-core/components/form-pill-group/stories/customization.stories.tsx:7]: nit(non-blocking): we can and probably should start all using the `Story` type from `@storybook/react` to type these story functions going forward.

File: packages/paste-core/primitives/box/src/types.ts
Comment (SiTaggart) [reply packages/paste-core/primitives/box/src/types.ts:93]: kind of amazed this wasn't already there. 🤯

File: packages/paste-design-tokens/tokens/global/text-color.yml
Comment (SiTaggart) [inline packages/paste-design-tokens/tokens/global/text-color.yml:271]: praise: Nice, keeping us accountable to color contrast

**Review by SiTaggart — APPROVED** [review]

**Review by nkrantz — APPROVED** [review]

**Review by gloriliale — APPROVED** [review]

**Pull request merged**
